client/pages/newTrip.jsx

```diff
@@ -0,0 +1,248 @@
+                    <label htmlFor="startDate">
+                      End Date
+                    </label>
+                  </div>
+                  <div>
+                    <input
+                      required
+                      id="endDate"
+                      type="date"
+                      name="endDate"
+                      onChange={this.handleChange}
+                      className="input poppins">
+                    </input>
+                  </div>
+                </div>
+                <div className="flex justify-center plan-button-margin">
+                  <button id="start-planning" type="submit" className="plan-button poppins">
+                    Start Planning
+                  </button>
+                </div>
+              </div>
+            </div>
+          </div>
+        </form>
+      </>
+      );
+    } else if (this.state.modalOpen === true) {
+      return (
+        <>
+          <div className="icon-modal margin-auto">
+            <div className="row justify-space-between flex-wrap">
+              <a><img id="barn" className="icon" onClick={this.handleClick} src="icons/icon-barn.svg"></img></a>
+              <a><img id="bay" className="icon" onClick={this.handleClick} src="icons/icon-bay.svg"></img></a>
+              <a><img id="beach" className="icon" onClick={this.handleClick} src="icons/icon-beach.svg"></img></a>
+              <a><img id="camp" className="icon" onClick={this.handleClick} src="icons/icon-camp.svg"></img></a>
+
+              <a><img id="canyon" className="icon" onClick={this.handleClick} src="icons/icon-canyon.svg"></img></a>
+              <a><img id="city" className="icon" onClick={this.handleClick} src="icons/icon-city.svg"></img></a>
+              <a><img id="country" className="icon" onClick={this.handleClick} src="icons/icon-country.svg"></img></a>
+              <a><img id="desert" className="icon" onClick={this.handleClick} src="icons/icon-desert.svg"></img></a>
+
+              <a><img id="industrial" className="icon" onClick={this.handleClick} src="icons/icon-industrial.svg"></img></a>
+              <a><img id="lake" className="icon" onClick={this.handleClick} src="icons/icon-lake.svg"></img></a>
+              <a><img id="lighthouse" className="icon" onClick={this.handleClick} src="icons/icon-lighthouse.svg"></img></a>
+              <a><img id="mountain-lake" className="icon" onClick={this.handleClick} src="icons/icon-mountain-lake.svg"></img></a>
+
+              <a><img id="mountain" className="icon" onClick={this.handleClick} src="icons/icon-mountain.svg"></img></a>
+              <a><img id="night-lake" className="icon" onClick={this.handleClick} src="icons/icon-night-lake.svg"></img></a>
+              <a><img id="plateau" className="icon" onClick={this.handleClick} src="icons/icon-plateau.svg"></img></a>
+              <a><img id="barn" className="icon" onClick={this.handleClick} src="icons/icon-rural.svg"></img></a>
+            </div>
```

Comment:
I recommend extracting this to an `IconPicker` function component with two props, `open` and `onClick`. If `props.open` is false, you return `null`, otherwise you return these elements. That way you don't need to copy-paste your form.